operate/common/src/main/java/io/camunda/operate/util/SoftHashMap.java

```diff
@@ -153,7 +153,7 @@ private void addToStrongReferences(V result) {
   private void trimStrongReferencesIfNecessary() {
     // trim the strong ref queue if necessary:
     while (strongReferences.size() > retentionSize) {
-      strongReferences.poll();
```

Comment:
⬆️ This was the acutal change. Sorry for the style updates.

Reply:
Curious, why did spotbugs pick this up and what is the benefit of remove() over poll() ? It looks like from the documentation the only main difference is remove() can throw an exception if the queue is empty whereas poll() returns null.

Reply:
I guess because it is often unintended to use poll to remove something from the queue and not do something with it. 

The checks likely check the different interfaces and such methods where this is not expected. But I see you point with removing the last causing an exception, but I think here it is not an issue.

Reply:
remove() also returns a value, would spotbugs still flag it afterwards since we still wouldn't be doing anything with a value?

Reply:
It wasn't flagging it, I guess there some rules to allow specific APIs. 

As mentioned earlier I think it is not an issue with the remove and empty collection (throwing exception), as we only do it if it is too packed, but I still understand your point that we change behavior.

I would close the PR for now. Feel free to do the change or ignore the bug flagged by spotbugs via an annotation. At the end it is your call, I'm not confident enough with operate code base to decide that.